From: Thomas Klute <thomas2.klute@uni-dortmund.de>
Date: Sun, 12 Feb 2017 13:24:54 +0100
Subject: Test suite: Do not explicitly set the mutex type to "default"

The setting can cause trouble when the mutex type "default" is file
based and its definition includes a path that the build process cannot
write to. This problem caused the Debian build to fail on hurd-i386,
where "default" resolved to "file:/var/run/apache2/" according to the
build log.

According to the HTTPD documentation a run-time file directory
relative to ServerRoot is used absent an explicit setting, and the
ServerRoot defined in test/base_apache.conf must be writable for the
test suite anyway.
---
 configure.ac                   | 8 ++++----
 test/apache-conf/netns.conf.in | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 425f2b8..cc3e8ae 100644
--- a/configure.ac
+++ b/configure.ac
@@ -152,11 +152,11 @@ AM_CONDITIONAL([ENABLE_NETNS], [test "$use_netns" != "no"])
 # Adjust Apache configuration for tests accordingly: Use pthread mutex
 # and test specific PID files if using namespaces, defaults otherwise.
 AS_IF([test "$use_netns" = "yes"],
-      [MUTEX_TYPE="pthread"; PID_AFFIX="-\${TEST_NAME}"],
-      [MUTEX_TYPE="default"; PID_AFFIX=""])
-AC_SUBST(MUTEX_TYPE)
+      [MUTEX_CONF="Mutex pthread default"; PID_AFFIX="-\${TEST_NAME}"],
+      [MUTEX_CONF=""; PID_AFFIX=""])
+AC_SUBST(MUTEX_CONF)
 AC_SUBST(PID_AFFIX)
-AM_SUBST_NOTMAKE(MUTEX_TYPE)
+AM_SUBST_NOTMAKE(MUTEX_CONF)
 AM_SUBST_NOTMAKE(PID_AFFIX)
 
 AC_ARG_ENABLE(msva,
diff --git a/test/apache-conf/netns.conf.in b/test/apache-conf/netns.conf.in
index 2439337..005d48f 100644
--- a/test/apache-conf/netns.conf.in
+++ b/test/apache-conf/netns.conf.in
@@ -1,4 +1,4 @@
 # This file contains options that are different depending on whether
 # tests use namespaces or not.
-Mutex	@MUTEX_TYPE@	default
+@MUTEX_CONF@
 PidFile	apache2@PID_AFFIX@.pid
